bundles/org.jkiss.utils/tests/org/jkiss/utils/BeanUtilsTest.java

```diff
@@ -0,0 +1,218 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.utils;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.powermock.api.mockito.PowerMockito;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.WildcardType;
+
+public class BeanUtilsTest {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testIsGetterName() {
+        Assert.assertFalse(BeanUtils.isGetterName(""));
+        Assert.assertFalse(BeanUtils.isGetterName("foo"));
+
+        Assert.assertTrue(BeanUtils.isGetterName("is"));
+        Assert.assertTrue(BeanUtils.isGetterName("get"));
+        Assert.assertTrue(BeanUtils.isGetterName("has"));
+    }
+
+    @Test
+    public void testGetPropertyNameFromGetter() {
+        Assert.assertNull(BeanUtils.getPropertyNameFromGetter("foobar"));
+
+        Assert.assertEquals("bar",
+                BeanUtils.getPropertyNameFromGetter("isbar"));
+        Assert.assertEquals("bar",
+                BeanUtils.getPropertyNameFromGetter("getbar"));
+        Assert.assertEquals("bar",
+                BeanUtils.getPropertyNameFromGetter("hasbar"));
+    }
+
+    @Test
+    public void testGetSetterName() {
+        Assert.assertNull(BeanUtils.getSetterName("foobar"));
+
+        Assert.assertEquals("setbar", BeanUtils.getSetterName("isbar"));
+        Assert.assertEquals("setbar", BeanUtils.getSetterName("getbar"));
+        Assert.assertEquals("setbar", BeanUtils.getSetterName("hasbar"));
+    }
+
+    @Test
+    public void testGetSetMethod() {
+        Assert.assertNull(BeanUtils.getSetMethod(String.class, "size"));
+        Assert.assertNull(BeanUtils.getSetMethod(String.class, "length"));
+        Assert.assertNull(
+                BeanUtils.getSetMethod(String.class, "length", true));
+        Assert.assertNull(
+                BeanUtils.getSetMethod(String.class, "length", false));
+    }
+
+    @Test
+    public void testGetGetMethod() {
+        Assert.assertNull(BeanUtils.getGetMethod(String.class, "size"));
+        Assert.assertNull(BeanUtils.getGetMethod(String.class, "length"));
+        Assert.assertNull(
+                BeanUtils.getGetMethod(String.class, "length", true));
+        Assert.assertNull(
+                BeanUtils.getGetMethod(String.class, "length", false));
+    }
+
```

Comment:
There is no test for `propertyNameToMethodName()`. Not sure if this was intentional as it looks coverable and other methods seems to be present.

Reply:
Ooops.. Done